packages/backend/apps/multitenancy/middleware.py

```diff
@@ -48,16 +40,44 @@ def get_current_user_role(info):
     return None
 
 
+def get_tenant_id_from_arguments(args):
+    """
+    Extract the tenant ID from GraphQL arguments.
+
+    Args:
+        args (dict): GraphQL arguments.
+
+    Returns:
+        str or None: The extracted tenant ID or None if not found.
+    """
+    request_input = args.get("input")
+    if request_input:
+        tenant_id = request_input.get("tenant_id")
+        if not tenant_id:
+            # for the purpose of Tenant CRUD actions
+            tenant_id = request_input.get("id")
+    else:
+        # for the purpose of queries, where is no input, just parameters
+        tenant_id = args.get("id")
+    if tenant_id:
+        id_type, pk = from_global_id(tenant_id)
+        if id_type == "TenantType":
+            return pk
+    return None
+
+
 class TenantUserRoleMiddleware(object):
```

Comment:
I'm thinking that at some point you need to add a middleware for django channels as well. Check out `JSONWebTokenCookieMiddleware` and asgi.py